pkg/operator/ceph/object/spec.go

```diff
@@ -288,7 +300,7 @@ func (c *clusterConfig) generateLiveProbeScheme() v1.URIScheme {
 
 	// If rgw is configured to use a secured port we need get on https://
 	// Only do this when the Non-SSL port is not used
-	if c.store.Spec.Gateway.Port == 0 && c.store.Spec.Gateway.SecurePort != 0 && c.store.Spec.Gateway.SSLCertificateRef != "" {
+	if c.store.Spec.Gateway.Port == 0 && c.store.Spec.Gateway.SecurePort != 0 && (c.store.Spec.Gateway.SSLCertificateRef != "" || len(getServiceServingCert(c.store.Spec.Gateway)) != 0) {
```

Comment:
How about a helper method such as this? It would help simplify the conditions that I see this repeated in several places.
```go
 func secureGatewayEnabled() bool {
  return c.store.Spec.Gateway.SecurePort != 0 && (c.store.Spec.Gateway.SSLCertificateRef != "" || len(getServiceServingCert(c.store.Spec.Gateway)) != 0)
}

Reply:
@travisn I have helper function for [sslcheck](https://github.com/rook/rook/pull/7331/commits/667cd47f287a86b396130ed2efea6c82bf333aca#r615556663) in "https://github.com/rook/rook/pull/7331" . If that PR merges I will rebase this one on top of it